app/admin/user.rb

```diff
@@ -104,17 +113,40 @@ def password_params
     @user.assign_attributes(password_params)
 
     if @user.save
-      ActiveAdminActionLog.create do |log|
-        log.user = current_user
-        log.resource = resource
-        log.path = resource_path
-        log.action = action_name
-      end
+      action_log(user: current_user,
+                 resource: resource,
+                 path: resource_path,
+                 action: action_name)
 
       redirect_to action: :index
     else
       flash[:error] = @user.errors.full_messages
       redirect_to :back
     end
   end
+
+  member_action :delete_monthly_report, method: :delete do
+    monthly_reports = resource.monthly_reports
+    if !resource.active_for_authentication? && monthly_reports.destroy_all
+      action_log(user: current_user,
+                 resource: resource,
+                 path: resource_path,
+                 action: action_name)
+
+      flash[:success] = '月報を削除しました'
+    else
+      flash[:error] = '月報の削除に失敗しました'
+    end
+
+    redirect_to :back
+  end
+
+  action_item :delete_monthly_report, only: :show do
+    if !user.active_for_authentication?
```

Comment:
__[rubocop]__
* Favor `unless` over `if` for negative conditions. [:ref](https://github.com/bbatsov/ruby-style-guide#unless-for-negatives)

<!-- Comment by [SideCI](https://sideci.com) -->